### PR TITLE
update docs on the required z3 version

### PR DESCRIPTION
**Pull request by leonardoalt**

### CODE REVIEW

**Review by r0qs — DISMISSED** [review]

File: docs/smtchecker.rst [hunk 1]
```diff
@@ -632,7 +632,8 @@ option ``--model-checker-solvers {all,cvc4,eld,smtlib2,z3}`` or the JSON option
 .. note::
   z3 version 4.8.16 broke ABI compatibility with previous versions and cannot
   be used with solc <=0.8.13. If you are using z3 >=4.8.16 please use solc
-  >=0.8.14.
+  >=0.8.14, and vice versa. We also recommend using the latest z3 release which
```
Comment (cameel) [inline docs/smtchecker.rst]: This vice versa reads like "if you are using solc >=0.8.14, please use z3 >= 4.8.16" to me, shouldn't it be this:

```suggestion
  >=0.8.14, and conversely, only use older z3 with older solc releases. We also recommend using the latest z3 release which
```

**Review by cameel — DISMISSED** [review]
LGTM.

I think the "vice versa" part needs to be reworded, but I guess it will still be understood so I'm approving anyway.

**Review by nikola-matic — DISMISSED** [review]

File: docs/smtchecker.rst [hunk 2]
```diff
@@ -632,7 +632,8 @@ option ``--model-checker-solvers {all,cvc4,eld,smtlib2,z3}`` or the JSON option
 .. note::
   z3 version 4.8.16 broke ABI compatibility with previous versions and cannot
   be used with solc <=0.8.13. If you are using z3 >=4.8.16 please use solc
-  >=0.8.14.
+  >=0.8.14, and conversely, only use older z3 with older solc releases.
```
Comment (leonardoalt) [inline docs/smtchecker.rst:635]: What I meant before was rather that if you're using Solidity >=0.8.14 you should also use the mentioned z3

Comment (nikola-matic) [reply docs/smtchecker.rst:635]: Personally, I was fine with the 'vice versa' version, however, this doesn't make the statement false, right? I.e. a pre `0.8.14` solc will not work with `4.8.16+` z3?

Comment (leonardoalt) [reply docs/smtchecker.rst:635]: Yep, what you added makes sense too. Maybe rephrase to something "please use Solc >=0.8.16 in combination with z3 v , and older solcs with z3 other v, resp."

Comment (nikola-matic) [reply docs/smtchecker.rst:635]:
```
  z3 version 4.8.16 broke ABI compatibility with previous versions and cannot
  be used with solc <=0.8.13. If you are using z3 >=4.8.16 please use solc
  >=0.8.14, and conversely, only use older z3 with older solc releases.
  We also recommend using the latest z3 release which is what SMTChecker also does.
```
Honestly, I this it's fine as is - especially given the context of `4.8.16` breaking the ABI. As in, what's written here makes perfect sense to me.

Comment (leonardoalt) [reply docs/smtchecker.rst:635]: Sounds good

**Review by nikola-matic — APPROVED** [review]
